docs/tools/firmware.sh

```diff
@@ -5,500 +5,789 @@
-				cp "$0" "/Volumes/${EFILABEL}/firmware.sh" 2>/dev/null || curl -s https://wiki.t2linux.org/tools/firmware.sh > "/Volumes/${EFILABEL}/firmware.sh" || (echo -e "\nFailed to copy script.\nPlease copy the script manually to the EFI partition using Finder\nMake sure the name of the script is firmware.sh in the EFI partition\n" && echo && read -p "Press enter after you have copied" && echo)
-				echo "Unmounting the EFI partition"
+				cp "$0" "/Volumes/${EFILABEL}/firmware.sh" 2>/dev/null \
+					|| curl -s https://wiki.t2linux.org/tools/firmware.sh > "/Volumes/${EFILABEL}/firmware.sh" \
+					|| (echo && cat <<- EOF && read -r
+						Failed to copy script.
+						Please copy the script manually to the EFI partition using Finder
+						Make sure the name of the script is firmware.sh in the EFI partition
+						Press enter after you have copied"
+						EOF
+						)
 				sudo diskutil unmount "/Volumes/${EFILABEL}/"
-				echo
-				echo -e "Run the following commands or run this script itself in Linux now to set up Wi-Fi :-\n\nsudo mkdir -p /tmp/apple-wifi-efi\nsudo mount /dev/nvme0n1p1 /tmp/apple-wifi-efi\nbash /tmp/apple-wifi-efi/firmware.sh\nsudo umount /tmp/apple-wifi-efi\n"
+				cat <<- EOF
+
+				Run the following commands or run this script itself in Linux now to set up Wi-Fi:
+
+				sudo mkdir -p /tmp/apple-wifi-efi
+				sudo mount /dev/nvme0n1p1 /tmp/apple-wifi-efi
+				bash /tmp/apple-wifi-efi/firmware.sh
+				sudo umount /tmp/apple-wifi-efi
+				EOF
 				;;
 			(2)
 
-				echo -e "\nChecking for missing dependencies"
-				python_check
 				echo -e "\nCreating a tarball of the firmware"
-				python3 "$0" /usr/share/firmware $HOME/Downloads/firmware.tar ${verbose}
-				if [[ (${identifier} = iMac19,1) || (${identifier} = iMac19,2) || (${identifier} = iMacPro1,1) ]]
-		then
-					nvramfile=$(ioreg -l | grep RequestedFiles | cut -d "/" -f 5 | rev | cut -c 4- | rev)
-					txcapblob=$(ioreg -l | grep RequestedFiles | cut -d "/" -f 3 | cut -d "\"" -f 1)
-					cp ${verbose} /usr/share/firmware/wifi/C-4364__s-B2/${nvramfile} "$HOME/Downloads/brcmfmac4364b2-pcie.txt"
-					cp ${verbose} /usr/share/firmware/wifi/C-4364__s-B2/${txcapblob} "$HOME/Downloads/brcmfmac4364b2-pcie.txcap_blob"
-					cd $HOME/Downloads
-					tar --append ${verbose} -f firmware.tar brcmfmac4364b2-pcie.txt
-					tar --append ${verbose} -f firmware.tar brcmfmac4364b2-pcie.txcap_blob
-					rm ${verbose} brcmfmac4364b2-pcie.txt
-					rm ${verbose} brcmfmac4364b2-pcie.txcap_blob
-					cd - >/dev/null
-				fi
-				echo -e "\nFirmware tarball saved to Downloads!"
-				echo -e "\nExtract the tarball contents to /lib/firmware/brcm in Linux and run the following in the Linux terminal:"
-				echo -e "\nsudo modprobe -r brcmfmac_wcc"
-				echo "sudo modprobe -r brcmfmac"
-				echo "sudo modprobe brcmfmac"
-				echo "sudo modprobe -r hci_bcm4377"
-				echo "sudo modprobe hci_bcm4377"
+				create_firmware_archive /usr/share/firmware "$HOME/Downloads/firmware.tar" ${verbose}
```

Comment:
You have already specified verbose in the create_firmware_archive function. Is it required here as well?

Reply:
Its a leftover from when I had `verbose=$3`, it doesnt do anything now. Im going to remove it